<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-08-18">

<title>WildTrax: a platform for the management, storage, processing, sharing and discovery of avian data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="wildtrax-ace-eco_files/libs/clipboard/clipboard.min.js"></script>
<script src="wildtrax-ace-eco_files/libs/quarto-html/quarto.js"></script>
<script src="wildtrax-ace-eco_files/libs/quarto-html/popper.min.js"></script>
<script src="wildtrax-ace-eco_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="wildtrax-ace-eco_files/libs/quarto-html/anchor.min.js"></script>
<link href="wildtrax-ace-eco_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="wildtrax-ace-eco_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="wildtrax-ace-eco_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="wildtrax-ace-eco_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="wildtrax-ace-eco_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="wildtrax-ace-eco_files/libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="wildtrax-ace-eco_files/libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="wildtrax-ace-eco_files/libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#abstract" id="toc-abstract" class="nav-link active" data-scroll-target="#abstract">Abstract</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#birds-as-ecological-indicators" id="toc-birds-as-ecological-indicators" class="nav-link" data-scroll-target="#birds-as-ecological-indicators">Birds as ecological indicators</a></li>
  <li><a href="#environmental-sensors-and-big-data" id="toc-environmental-sensors-and-big-data" class="nav-link" data-scroll-target="#environmental-sensors-and-big-data">Environmental sensors and big data</a></li>
  <li><a href="#open-avian-data" id="toc-open-avian-data" class="nav-link" data-scroll-target="#open-avian-data">Open avian data</a></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#database" id="toc-database" class="nav-link" data-scroll-target="#database">Database</a>
  <ul class="collapse">
  <li><a href="#infrastructure" id="toc-infrastructure" class="nav-link" data-scroll-target="#infrastructure">Infrastructure</a></li>
  <li><a href="#schemas" id="toc-schemas" class="nav-link" data-scroll-target="#schemas">Schemas</a></li>
  </ul></li>
  <li><a href="#front-end" id="toc-front-end" class="nav-link" data-scroll-target="#front-end">Front-end</a>
  <ul class="collapse">
  <li><a href="#user-interface-ui-user-experience-ux-and-content-management-system-cms" id="toc-user-interface-ui-user-experience-ux-and-content-management-system-cms" class="nav-link" data-scroll-target="#user-interface-ui-user-experience-ux-and-content-management-system-cms">User interface (UI), user experience (UX) and content management system (CMS)</a></li>
  </ul></li>
  <li><a href="#data-management-and-processing" id="toc-data-management-and-processing" class="nav-link" data-scroll-target="#data-management-and-processing">Data management and processing</a>
  <ul class="collapse">
  <li><a href="#acoustic-sensor" id="toc-acoustic-sensor" class="nav-link" data-scroll-target="#acoustic-sensor">Acoustic sensor</a></li>
  <li><a href="#camera-sensor" id="toc-camera-sensor" class="nav-link" data-scroll-target="#camera-sensor">Camera sensor</a></li>
  <li><a href="#avian-point-counts" id="toc-avian-point-counts" class="nav-link" data-scroll-target="#avian-point-counts">Avian point counts</a></li>
  </ul></li>
  <li><a href="#data-publication-sharing-and-discovery" id="toc-data-publication-sharing-and-discovery" class="nav-link" data-scroll-target="#data-publication-sharing-and-discovery">Data publication, sharing and discovery</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#growth-and-community" id="toc-growth-and-community" class="nav-link" data-scroll-target="#growth-and-community">Growth and community</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  <li><a href="#additional-information-and-declarations" id="toc-additional-information-and-declarations" class="nav-link" data-scroll-target="#additional-information-and-declarations">Additional Information and Declarations</a>
  <ul class="collapse">
  <li><a href="#competing-interests" id="toc-competing-interests" class="nav-link" data-scroll-target="#competing-interests">Competing Interests</a></li>
  <li><a href="#author-contributions" id="toc-author-contributions" class="nav-link" data-scroll-target="#author-contributions">Author Contributions</a></li>
  </ul></li>
  <li><a href="#acknowledgements" id="toc-acknowledgements" class="nav-link" data-scroll-target="#acknowledgements">Acknowledgements</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">WildTrax: a platform for the management, storage, processing, sharing and discovery of avian data</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>Alexander G. MacPhail </p>
             <p>Corrina Copp </p>
             <p>Erin M. Bayne </p>
             <p>Michael Packer </p>
             <p>Chad Klassen </p>
             <p>Elly C. Knight </p>
             <p>Steven L. Van Wilgenburg </p>
             <p>Kevin G. Kelly </p>
             <p>Joan Fang </p>
             <p>Monica Kohler </p>
             <p>Tara Narwani </p>
             <p>Hedwig E. Lankau </p>
             <p>Charles M. Francis </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 18, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/data-discover.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="assets/data-discover.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="float:left; border:1px solid #ccc; padding:4px;" alt="Data Discover"></a></p>
</figure>
</div>
<section id="abstract" class="level1">
<h1>Abstract</h1>
<p>As environmental sensors become essential tools for monitoring and assessing bird population trends, their effective use relies on robust systems to manage the large, and sometimes overwhelming, datasets they generate. WildTrax (https://www.wildtrax.ca) is a web-based platform designed to manage, store, process, share, and discover environmental sensor data across local to international extents. By creating, developing and mainting this software, WildTrax enables researchers to address ecological questions at multiple spatial and temporal scales through novel analytical approaches, strengthening the avian data network, fostering collaboration, and enhancing data sharing to support bird conservation across Canada and beyond.</p>
<div class="callout callout-style-simple callout-none no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Highlights</strong></p>
<ul>
<li>Environmental sensor data generate large and complex datasets, requiring structured systems for harmonization, quality control, and integration across studies.</li>
<li>WildTrax provides a framework for storing, managing, processing, sharing, and discovering ecological data, enabling broader accessibility and reproducibility.</li>
<li>Analytical pipelines, including thoughtfully applied artificial intelligence tools, can enhance data interpretation while maintaining transparency and user control.</li>
<li>Continued development of integrated data platforms fosters opportunities for researchers to synthesize diverse datasets, advancing both ecological understanding and conservation practice.</li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-none no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Keywords:</strong><br>
Environmental sensors, bird population monitoring, data management, online platform, ecological data sharing, multi-scale analysis, conservation technology</p>
</div>
</div>
</div>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="birds-as-ecological-indicators" class="level2">
<h2 class="anchored" data-anchor-id="birds-as-ecological-indicators">Birds as ecological indicators</h2>
<p>Birds have long been recognized as ecological indicators (<span class="citation" data-cites="temple1989bird">Temple and Wiens (<a href="#ref-temple1989bird" role="doc-biblioref">1989</a>)</span>, <span class="citation" data-cites="bibby1999making">Bibby (<a href="#ref-bibby1999making" role="doc-biblioref">1999</a>)</span>, <span class="citation" data-cites="canterbury2000bird">Canterbury et al. (<a href="#ref-canterbury2000bird" role="doc-biblioref">2000</a>)</span>, <span class="citation" data-cites="gregory2003using">Gregory et al. (<a href="#ref-gregory2003using" role="doc-biblioref">2003</a>)</span>, <span class="citation" data-cites="mekonen2017birds">Mekonen (<a href="#ref-mekonen2017birds" role="doc-biblioref">2017</a>)</span>) due to their sensitivity to environmental changes (<span class="citation" data-cites="furness2013birds">Furness and Greenwood (<a href="#ref-furness2013birds" role="doc-biblioref">2013</a>)</span>), broad distribution across ecosystems (<span class="citation" data-cites="orme2006global">Orme et al. (<a href="#ref-orme2006global" role="doc-biblioref">2006</a>)</span>, <span class="citation" data-cites="jetz2012global">Jetz et al. (<a href="#ref-jetz2012global" role="doc-biblioref">2012</a>)</span>, <span class="citation" data-cites="aide2013real">Aide et al. (<a href="#ref-aide2013real" role="doc-biblioref">2013</a>)</span>, <span class="citation" data-cites="lepage2014avibase">Lepage, Vaidya, and Guralnick (<a href="#ref-lepage2014avibase" role="doc-biblioref">2014</a>)</span>), and measurable population dynamics (<span class="citation" data-cites="reif2013long">Reif (<a href="#ref-reif2013long" role="doc-biblioref">2013</a>)</span>). They can signal shifts in ecosystem health (<span class="citation" data-cites="zakaria2005comparison">Zakaria, Leong, and Yusuf (<a href="#ref-zakaria2005comparison" role="doc-biblioref">2005</a>)</span>, <span class="citation" data-cites="newman2007aquatic">Newman et al. (<a href="#ref-newman2007aquatic" role="doc-biblioref">2007</a>)</span>, <span class="citation" data-cites="smits2013avian">Smits and Fernie (<a href="#ref-smits2013avian" role="doc-biblioref">2013</a>)</span>) and function (<span class="citation" data-cites="sekercioglu2012bird">Sekercioglu (<a href="#ref-sekercioglu2012bird" role="doc-biblioref">2012</a>)</span>, <span class="citation" data-cites="sitters2016bird">Sitters et al. (<a href="#ref-sitters2016bird" role="doc-biblioref">2016</a>)</span>, <span class="citation" data-cites="morante2017appraisal">Morante-Filho and Faria (<a href="#ref-morante2017appraisal" role="doc-biblioref">2017</a>)</span>), providing integral metrics (<span class="citation" data-cites="stanton2016flexible">Stanton et al. (<a href="#ref-stanton2016flexible" role="doc-biblioref">2016</a>)</span>, <span class="citation" data-cites="michel2020metrics">Michel et al. (<a href="#ref-michel2020metrics" role="doc-biblioref">2020</a>)</span>) for conservation and management. Birds are also valuable indicators because they occupy diverse niches, have well-studied life histories, and often correlate with the health of other taxa (<span class="citation" data-cites="fleishman2005using">Fleishman et al. (<a href="#ref-fleishman2005using" role="doc-biblioref">2005</a>)</span>). Their utility spans ecosystem monitoring, habitat quality assessment, and gauging the impact of environmental stressors such as land use change, climate change (<span class="citation" data-cites="jiguet2007climate">Jiguet et al. (<a href="#ref-jiguet2007climate" role="doc-biblioref">2007</a>)</span>), and pollution (<span class="citation" data-cites="niemi1997critical">Niemi et al. (<a href="#ref-niemi1997critical" role="doc-biblioref">1997</a>)</span>, <span class="citation" data-cites="mekonen2017birds">Mekonen (<a href="#ref-mekonen2017birds" role="doc-biblioref">2017</a>)</span>). Advances in statistical modelling now allow for better integration of uncertainty, phylogenetic relationships, and temporal autocorrelation, enhancing the reliability of bird-based indicators (<span class="citation" data-cites="fraixedas2020state">Fraixedas et al. (<a href="#ref-fraixedas2020state" role="doc-biblioref">2020</a>)</span>). However, there remains challenges in their application as indicators, including spatial, seasonal, and habitat biases, as well as insufficient consideration of statistical uncertainty and temporal autocorrelation in multi-species bird indicators (<span class="citation" data-cites="gregory2003using">Gregory et al. (<a href="#ref-gregory2003using" role="doc-biblioref">2003</a>)</span>, <span class="citation" data-cites="fraixedas2020state">Fraixedas et al. (<a href="#ref-fraixedas2020state" role="doc-biblioref">2020</a>)</span>).</p>
</section>
<section id="environmental-sensors-and-big-data" class="level2">
<h2 class="anchored" data-anchor-id="environmental-sensors-and-big-data">Environmental sensors and big data</h2>
<p>Environmental sensors, such as autonomous recording units (ARUs) and remote camera traps, are reshaping avian monitoring by providing continuous, high-resolution, and large-scale data collection (<span class="citation" data-cites="hobson2002acoustic">Hobson et al. (<a href="#ref-hobson2002acoustic" role="doc-biblioref">2002</a>)</span>, <span class="citation" data-cites="shonfield2017autonomous">Shonfield and Bayne (<a href="#ref-shonfield2017autonomous" role="doc-biblioref">2017</a>)</span>, <span class="citation" data-cites="pollet2025technological">Pollet et al. (<a href="#ref-pollet2025technological" role="doc-biblioref">2025</a>)</span>). The adoption of ARUs has allowed single-visit human surveys to be supplemented or replaced with archived acoustic recordings, enabling “big data” approaches that integrate multiple datasets for broader ecological inference (<span class="citation" data-cites="hampton2013big">Hampton et al. (<a href="#ref-hampton2013big" role="doc-biblioref">2013</a>)</span>; <span class="citation" data-cites="farley2018situating">Farley et al. (<a href="#ref-farley2018situating" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="shin2015ecological">Shin and Choi (<a href="#ref-shin2015ecological" role="doc-biblioref">2015</a>)</span>; <span class="citation" data-cites="nathan2022big">Nathan et al. (<a href="#ref-nathan2022big" role="doc-biblioref">2022</a>)</span>; <span class="citation" data-cites="peters2014harnessing">Peters et al. (<a href="#ref-peters2014harnessing" role="doc-biblioref">2014</a>)</span>; <span class="citation" data-cites="hallgren2016biodiversity">Hallgren et al. (<a href="#ref-hallgren2016biodiversity" role="doc-biblioref">2016</a>)</span>) while still being statistical harmonized across methodologies (<span class="citation" data-cites="solymos2013calibrating">Sólymos et al. (<a href="#ref-solymos2013calibrating" role="doc-biblioref">2013</a>)</span>) with detectability rates depend on factors such as distance from the ARU, frequency range, and habitat structure (<span class="citation" data-cites="yip2017experimentally">Yip et al. (<a href="#ref-yip2017experimentally" role="doc-biblioref">2017</a>)</span>). Remote camera traps have likewise become valuable tools for avian research, particularly for documenting nesting behavior, predation events, and species presence in otherwise inaccessible habitats (<span class="citation" data-cites="bolton2007remote">Bolton et al. (<a href="#ref-bolton2007remote" role="doc-biblioref">2007</a>)</span>; <span class="citation" data-cites="randler2018distance">Randler and Kalb (<a href="#ref-randler2018distance" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="o2008picture">O’Brien and Kinnaird (<a href="#ref-o2008picture" role="doc-biblioref">2008</a>)</span>). When deployed alongside ARUs, cameras provide complementary visual evidence that strengthens ecological inference, highlighting the advantages of multimodal sensor programs (<span class="citation" data-cites="buxton2018pairing">Buxton et al. (<a href="#ref-buxton2018pairing" role="doc-biblioref">2018</a>)</span>; <span class="citation" data-cites="garland2020acoustic">Garland et al. (<a href="#ref-garland2020acoustic" role="doc-biblioref">2020</a>)</span>). Both sensors generate large ecological media datasets, often characterized as ‘big data’ due to their volume, variety (e.g., different file types), veracity (uncertainty in data quality such as noise or misclassification), and velocity (rate of accumulation or acquistion, <span class="citation" data-cites="hampton2013big">Hampton et al. (<a href="#ref-hampton2013big" role="doc-biblioref">2013</a>)</span>; <span class="citation" data-cites="farley2018situating">Farley et al. (<a href="#ref-farley2018situating" role="doc-biblioref">2018</a>)</span>). Centralized software platforms facilitate standardization, integration, and sharing of these data, enabling interdisciplinary collaboration and advancing biodiversity research, and helping to reduce bias (<span class="citation" data-cites="peters2014harnessing">Peters et al. (<a href="#ref-peters2014harnessing" role="doc-biblioref">2014</a>)</span>). Notably, machine learning is being combined with ecological models to meet computational demands, positioning sensors and big data systems to link raw ecological observations with actionable conservation strategies and sustainability goals (<span class="citation" data-cites="nathan2022big">Nathan et al. (<a href="#ref-nathan2022big" role="doc-biblioref">2022</a>)</span>) and minimizing data waste (<span class="citation" data-cites="binley2023minimizing">Binley et al. (<a href="#ref-binley2023minimizing" role="doc-biblioref">2023</a>)</span>).</p>
</section>
<section id="open-avian-data" class="level2">
<h2 class="anchored" data-anchor-id="open-avian-data">Open avian data</h2>
<p>To maximize their utility in an open framework, biodiversity data must align with FAIR principles: findable, accessible, interoperable, and reusable (<span class="citation" data-cites="kush2020fair">Kush et al. (<a href="#ref-kush2020fair" role="doc-biblioref">2020</a>)</span>). However, challenges persist around data quality, equitable access, and long-term preservation, highlighting the importance of strong and lasting socio-technical frameworks (<span class="citation" data-cites="shin2015ecological">Shin and Choi (<a href="#ref-shin2015ecological" role="doc-biblioref">2015</a>)</span>). Software platforms that implement these principles and harmonize diverse datasets while maintaining consistency, equity, and quality can unlock advanced species- and community-level analyses, ultimately transforming raw observations into actionable insights for conservation and policy (<span class="citation" data-cites="stephenson2020inventory">Stephenson and Stengel (<a href="#ref-stephenson2020inventory" role="doc-biblioref">2020</a>)</span>; <span class="citation" data-cites="fox2017generating">Fox et al. (<a href="#ref-fox2017generating" role="doc-biblioref">2017</a>)</span>). By overcoming limitations of traditional methods and promoting open data-sharing, such platforms enhance understanding of ecological trends (<span class="citation" data-cites="buxton2021key">Buxton et al. (<a href="#ref-buxton2021key" role="doc-biblioref">2021</a>)</span>) and foster collaboration among stakeholders, aligning priorities across regional to global scales (<span class="citation" data-cites="kartez2008information">Kartez and Casto (<a href="#ref-kartez2008information" role="doc-biblioref">2008</a>)</span>). Here we introduce WildTrax, a platform for storing, managing, processing, sharing, and discovering avian environmental sensor data. This paper describes version 2.0 of WildTrax, corresponding with its 2025 release.</p>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<section id="database" class="level2">
<h2 class="anchored" data-anchor-id="database">Database</h2>
<section id="infrastructure" class="level3">
<h3 class="anchored" data-anchor-id="infrastructure">Infrastructure</h3>
<p>WildTrax uses PostgreSQL, a free and open-source relational database management system (RDBMS). PostgreSQL is well-suited for managing complex biodiversity datasets and rich metadata that accompanies ARU and remote camera data, offering optimized querying, relational structure, and robust storage capabilities (<span class="citation" data-cites="douglas2003postgresql">Douglas and Douglas (<a href="#ref-douglas2003postgresql" role="doc-biblioref">2003</a>)</span>; <span class="citation" data-cites="zhang2009efficiently">Zhang, Gertz, and Gruenwald (<a href="#ref-zhang2009efficiently" role="doc-biblioref">2009</a>)</span>; <span class="citation" data-cites="kim2021ecobank">Kim et al. (<a href="#ref-kim2021ecobank" role="doc-biblioref">2021</a>)</span>). The application infrastructure runs on a virtualized server environment configured for dedicated applications. This architecture supports both daily user interactions and more computationally intensive batch tasks, such as uploading large volumes of audio and image files or running artificial intelligence models for species recognition. To ensure reliability and performance at scale, the system employs load balancing and redundancy measures, enabling near-continuous availability for a large and growing user base. The production server is hosted at the University of Alberta (Edmonton, Canada), with optional long-term storage services available both at the University of Alberta server or through Amazon Web Services (AWS) with nodes available in Montreal and Oregon. Archival off-site storage uses <a href="https://aws.amazon.com/s3/storage-classes/glacier/">AWS Deep Glacier</a> providing a recommended 3-2-1 data backup policy (<span class="citation" data-cites="perkel2019ways">Perkel (<a href="#ref-perkel2019ways" role="doc-biblioref">2019</a>)</span>). To optimize performance, a local M.2 drive paired with a CPU significantly reduces I/O bottlenecks, enabling data read / write speeds multiple times faster than traditional SSD NAS configurations. In practice, these enhancements improve query response times and batch processing speeds, while the upgraded CPU delivers smoother responsiveness and improved multitasking during high-demand operations. Media files are meticulously indexed and linked to the stored media depending on the storage location an Organization chooses.</p>
</section>
<section id="schemas" class="level3">
<h3 class="anchored" data-anchor-id="schemas">Schemas</h3>
<p>WildTrax employs a hierarchically structure to maintain data integrity, enforce standardization, and support interoperability of environmental sensors. The system implements three primary data schemas, ARUs, cameras, and point counts, organized into Projects, which act as centralized containers for all observations and sensor outputs associated with a specific study or research question. Organizations sit above Projects in the hierarchy, aggregating multiple Projects and unifying sensor data and media under a consistent framework.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/aru-projects.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="ARU project dashboard"><img src="assets/aru-projects.png" class="img-fluid figure-img" alt="ARU project dashboard"></a></p>
<figcaption>ARU project dashboard</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/pc-projects.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Point count project dashboard"><img src="assets/pc-projects.png" class="img-fluid figure-img" alt="Point count project dashboard"></a></p>
<figcaption>Point count project dashboard</figcaption>
</figure>
</div>
<p>Within this structure, all media are explicitly linked to their parent Location, the precise geographic point of deployment, ensuring that sensors and their outputs are inseparable from their environmental context. The schema enforces referential integrity while also streamlining front-end operations such as data retrieval, filtering, and visualization using optimized queries. Beyond internal consistency, it supports interoperability with external repositories like <a href="https://weatherstack.com/">Weatherstack</a> for historical meteorological records, as well as the integration of machine-learning–based acoustic classifiers such as BirdNET (<span class="citation" data-cites="kahl2021birdnet">Kahl et al. (<a href="#ref-kahl2021birdnet" role="doc-biblioref">2021</a>)</span>) and HawkEars (<span class="citation" data-cites="huus2025hawkears">Huus et al. (<a href="#ref-huus2025hawkears" role="doc-biblioref">2025</a>)</span>). By aligning temporal metadata (<code>YYYY-MM-DD HH:MM:SS</code>) with spatial coordinates (longitude, latitude, WGS84), the framework enables multi-dimensional analyses that connect raw sensor output with ecological, climatic, and biogeographic patterns.</p>
</section>
</section>
<section id="front-end" class="level2">
<h2 class="anchored" data-anchor-id="front-end">Front-end</h2>
<section id="user-interface-ui-user-experience-ux-and-content-management-system-cms" class="level3">
<h3 class="anchored" data-anchor-id="user-interface-ui-user-experience-ux-and-content-management-system-cms">User interface (UI), user experience (UX) and content management system (CMS)</h3>
<p>WildTrax’s user interface is a responsive web application built with <a href="https://vuejs.org/,%20version%203">Vuejs</a>, leveraging its modular component-based architecture, including the Composition API, for scalability, maintainability, and efficient logic reuse. The user interface (UI) is styled and enhanced using <a href="https://primevue.org/">PrimeVue</a>, among other libraries, which delivers rich, customizable components such as data tables and dashboards for seamless data exploration and visualization. The application is deployed via an Apache HTTP Server, which serves the front-end and handles API routing through reverse-proxy configurations to back-end services. WildTrax exposes APIs for data exchange and provides export tools in standard scientific formats (e.g., CSV, JSON, text and zip), supporting downstream integration with statistical and geospatial workflows.</p>
<p>Dashboards are designed to give users a clear, intuitive overview of their sensors’ data, enabling both novice and expert users to navigate and interpret project information with minimal friction. Key features, including filters, sorting, clearly labeled column headers with hover information, and tooltips, are complemented by dropdown menus accessed through responsive, context-aware controls, allowing users to refine or manipulate large datasets quickly without navigating complex database relationship structures.</p>
<p><img src="assets/aru-upload.png" class="img-fluid"></p>
<p>Visual elements such as sortable data tables, progress indicators, and status icons support efficient user scanning and pattern recognition, while maintaining consistency with the broader WildTrax design system. Attention to micro-interactions, such as inline feedback and notifications when data are updated or filters applied, reinforces a sense of responsiveness and reduces cognitive user load. The dashboard prioritizes transparency by linking each dataset directly back to its associated location and media, ensuring users can trace results from summary views down to the raw sensor files from Tasks to Organizations.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/context-menu-public.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="ARU context menu for a public project"><img src="assets/context-menu-public.png" class="img-fluid figure-img" alt="ARU context menu for a public project"></a></p>
<figcaption>ARU context menu for a public project</figcaption>
</figure>
</div>
<p>WildTrax 2.0 incorporates a more content management system (CMS) that provides support for translation (currently supported in English and French), internationalization (i18n), and content localization, ensuring that the platform can accommodate multilingual users and diverse regional requirements.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/cms.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Content management system"><img src="assets/cms.png" class="img-fluid figure-img" alt="Content management system"></a></p>
<figcaption>Content management system</figcaption>
</figure>
</div>
</section>
</section>
<section id="data-management-and-processing" class="level2">
<h2 class="anchored" data-anchor-id="data-management-and-processing">Data management and processing</h2>
<section id="acoustic-sensor" class="level3">
<h3 class="anchored" data-anchor-id="acoustic-sensor">Acoustic sensor</h3>
<p>Uploaded recordings supported are WAC and W4V (<a href="https://www.wildlifeacoustics.com/">Wildlife Acoustics</a> proprietary lossless compressed file types), FLAC (open-source lossless compressed file types, https://github.com/xiph/flac), MP3 (lossy compressed audio; (MPEG-1 Audio Layer III)) and WAV (uncompressed audio). Media files, except mp3, are converted to lossless audio codec, FLAC, that preserves bit-for-bit fidelity while reducing storage needs by approximately 50–70% compared to WAV. FLAC also allows researchers to uncompress the lossless data for use in other applications, avoided any issues that arise with lossy data compression (<span class="citation" data-cites="macphail2024audio">MacPhail et al. (<a href="#ref-macphail2024audio" role="doc-biblioref">2024</a>)</span>). Once audio recordings are archived, they are then converted into spectrograms using short-time Fourier transforms (STFT) using the <a href="https://sourceforge.net/projects/sox/">SoX</a>. Default project parameters (X-scale for duration of time and Y-scale for spectrogram height in pixels) define the size of the spectrogram but that can changed to modify the dimensions and range of the spectral signatures in order to isolate frequency ranges. These spectrograms then serve as inputs for algorithms for the tags.</p>
<p><a href="assets/dynamic-spectrogram-rugr.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="assets/dynamic-spectrogram-rugr.png" class="img-fluid"></a></p>
<p><img src="assets/aru-upload.png" class="img-fluid"></p>
<p>WildTrax provides an integrated interface for developing, testing, and applying automated species identification models based on machine learning algorithms, including Hidden Markov Models (HMMs) and other classifiers.</p>
<p>The interface comprises four primary components:</p>
<ul>
<li><p>Visualizer – Allows users to view, listen to, and annotate acoustic and image data. It supports multiple data formats and devices, with tools for zooming, filtering, and navigating recordings.</p></li>
<li><p>Species Validation – Enables users to label species presence or absence in recordings, creating validation datasets to assess model accuracy. This module also allows review and correction of ROIs generated automatically.</p></li>
</ul>
</section>
<section id="camera-sensor" class="level3">
<h3 class="anchored" data-anchor-id="camera-sensor">Camera sensor</h3>
<p>Camera users can therefore upload, process their image sets with AI, tag the remaining detail, conduct species verification</p>
</section>
<section id="avian-point-counts" class="level3">
<h3 class="anchored" data-anchor-id="avian-point-counts">Avian point counts</h3>
<p>The point count sensor serves as a repository for the <a href="borealbirds.ca">Boreal Avian Modelling Centre’s</a> point count data. Point counts are harmonized to a specific distance bands which methods are correspondent to ARU methods for synthesis. When exported ARU data can be setup in a point count format example ARU -&gt; 0-INF distance band with the ability to format correctable offsets using <code>wt_qpad_offsets()</code> in the wildrtrax package following methods by <span class="citation" data-cites="solymos2013calibrating">Sólymos et al. (<a href="#ref-solymos2013calibrating" role="doc-biblioref">2013</a>)</span>.</p>
</section>
</section>
<section id="data-publication-sharing-and-discovery" class="level2">
<h2 class="anchored" data-anchor-id="data-publication-sharing-and-discovery">Data publication, sharing and discovery</h2>
<p>When the project is completed and all Tasks processed, the status of the project can be changed to a published status. Project publication allows other WildTrax users, who are not project members, to access the media, metadata or species detections from the project either through the project context menus &gt; Download Report or <a href="https://discover.wildtrax.ca/">Data Discover</a>. The publication status will control how data will become visible across the system. Project publication will lock users from editing species detections and is considered the final version of the data.</p>
<p>Data Discover is the central hub for exploring environmental sensor data in WildTrax. In Data Discover, users and the public can search for data from ARUs, cameras, and point counts, using a variety of attribute filters, and create summary statistics within a dynamic mapping interface. Users can gain a comprehensive understanding of environmental sensor data in an area that interests them noting which organizations have published data on WildTrax, which species were detected and to what frequency, and explore media elements such as images and sounds captured in the environment.</p>
<p><img src="assets/data-discover1.png" class="img-fluid"></p>
<p><img src="assets/data-discover2.png" class="img-fluid"></p>
<p><img src="assets/data-discover3.png" class="img-fluid"></p>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="growth-and-community" class="level2">
<h2 class="anchored" data-anchor-id="growth-and-community">Growth and community</h2>
<p>The sheer growth of media added to the overtime seen in Figure shows how WildTrax and the media therein an exponential curve in the data.</p>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
</section>
<section id="additional-information-and-declarations" class="level1">
<h1>Additional Information and Declarations</h1>
<section id="competing-interests" class="level2">
<h2 class="anchored" data-anchor-id="competing-interests">Competing Interests</h2>
<p>The authors declare no competing interests.</p>
</section>
<section id="author-contributions" class="level2">
<h2 class="anchored" data-anchor-id="author-contributions">Author Contributions</h2>
<p>All authors contributed to manuscript edits and revisions.</p>
<p>Alexander G. MacPhail led the design of the manuscript, acoustic processing platform, and workflows, conducted data analysis, drafted the manuscript, and served as primary liaison for project coordination and user support.</p>
<p>Corrina Copp supported platform development and data management, with a particular focus on camera sensor design and development.</p>
<p>Erin M. Bayne made formative and ongoing contributions to the conceptual development, study design, and overall scientific guidance of the project.</p>
<p>Michael Packer served as lead developer, overseeing server architecture, back-end development, system scalability, user support, and ensuring robust data processing workflows.</p>
<p>Chad Klassen designed the user interface, provided front-end development, user support, and led user experience design to optimize accessibility and usability across devices.</p>
<p>Elly C. Knight provided data standardization, integration and analytical support especially with acoustic classifiers.</p>
<p>Kevin G. Kelly provided design implementation and testing especially with acoustic classifiers, as well as user support.</p>
<p>Steve L. Van Wilgenburg provided immeasurable feedback through the creation and use of the system at scale.</p>
<p>Joan Fang helped with the initial concept and development.</p>
<p>Monica Kohler provided support during initial funding and conceptual stages.</p>
<p>Tara Narwani provided support during initial funding and conceptual stages.</p>
<p>Hedwig E. Lankau contributed to the development of the Bioacoustic Information System, which provided the foundational framework for the design of the acoustic processing system.</p>
<p>Charles M. Francis provided support, inspiration, and advocacy, which helped secure the funding that made this WildTrax possible. His guidance was instrumental in initiating WildTrax and in shaping earlier work with Avichorus, which served as an inspiration for the platform.</p>
</section>
</section>
<section id="acknowledgements" class="level1">
<h1>Acknowledgements</h1>
<p>This project was supported by funding from Environment and Climate Change Canada, Alberta Environment and Parks, the Oil Sands Monitoring Program, and Canada’s Oil Sands Innovation Alliance. We also extend our gratitude to the following organizations for their partnership and support: U of A Sound Studies Institute, Joint Canada-Alberta Implementation Plan for Oil Sands Monitoring, InnoTech Alberta, University of Alberta, NSERC (Natural Sciences and Engineering Research Council of Canada), PTAC, Devon Energy, ConocoPhillips, Cenovus Energy, Nexen, Imperial Oil, Shell, Suncor Energy, Alberta Pacific Forest Industries Inc., Canadian Natural, Alberta Conservation Association, Parks Canada, University of Alberta, Government of Saskatchewan, and Ɂehdzo Got’ı̨nę Gots’ę́ Nákedı Sahtú Renewable Resources Board. We thank the community of users who continue to make WildTrax a strong guiding force in avian conservation.</p>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-aide2013real" class="csl-entry" role="listitem">
Aide, T Mitchell, Carlos Corrada-Bravo, Marconi Campos-Cerqueira, Carlos Milan, Giovany Vega, and Rafael Alvarez. 2013. <span>“Real-Time Bioacoustics Monitoring and Automated Species Identification.”</span> <em>PeerJ</em> 1: e103.
</div>
<div id="ref-bibby1999making" class="csl-entry" role="listitem">
Bibby, Colin J. 1999. <span>“Making the Most of Birds as Environmental Indicators.”</span> <em>Ostrich</em> 70 (1): 81–88.
</div>
<div id="ref-binley2023minimizing" class="csl-entry" role="listitem">
Binley, Allison D, Brandon PM Edwards, Gabriel Dansereau, Elly C Knight, and Iman Momeni-Dehaghi. 2023. <span>“Minimizing Data Waste.”</span> <em>Bulletin of the Ecological Society of America</em> 104 (2): 1–11.
</div>
<div id="ref-bolton2007remote" class="csl-entry" role="listitem">
Bolton, Mark, Nigel Butcher, Fiona Sharpe, Danaë Stevens, and Gareth Fisher. 2007. <span>“Remote Monitoring of Nests Using Digital Camera Technology.”</span> <em>Journal of Field Ornithology</em> 78 (2): 213–20.
</div>
<div id="ref-buxton2021key" class="csl-entry" role="listitem">
Buxton, Rachel T, Joseph R Bennett, Andrea J Reid, Charles Shulman, Steven J Cooke, Charles M Francis, Elizabeth A Nyboer, et al. 2021. <span>“Key Information Needs to Move from Knowledge to Action for Biodiversity Conservation in Canada.”</span> <em>Biological Conservation</em> 256: 108983.
</div>
<div id="ref-buxton2018pairing" class="csl-entry" role="listitem">
Buxton, Rachel T, Patrick E Lendrum, Kevin R Crooks, and George Wittemyer. 2018. <span>“Pairing Camera Traps and Acoustic Recorders to Monitor the Ecological Impact of Human Disturbance.”</span> <em>Global Ecology and Conservation</em> 16: e00493.
</div>
<div id="ref-canterbury2000bird" class="csl-entry" role="listitem">
Canterbury, Grant E, Thomas E Martin, Daniel R Petit, Lisa J Petit, and David F Bradford. 2000. <span>“Bird Communities and Habitat as Ecological Indicators of Forest Condition in Regional Monitoring.”</span> <em>Conservation Biology</em> 14 (2): 544–58.
</div>
<div id="ref-douglas2003postgresql" class="csl-entry" role="listitem">
Douglas, Korry, and Susan Douglas. 2003. <em>PostgreSQL: A Comprehensive Guide to Building, Programming, and Administering PostgresSQL Databases</em>. SAMS publishing.
</div>
<div id="ref-farley2018situating" class="csl-entry" role="listitem">
Farley, Scott S, Andria Dawson, Simon J Goring, and John W Williams. 2018. <span>“Situating Ecology as a Big-Data Science: Current Advances, Challenges, and Solutions.”</span> <em>BioScience</em> 68 (8): 563–76.
</div>
<div id="ref-fleishman2005using" class="csl-entry" role="listitem">
Fleishman, Erica, James R Thomson, Ralph Mac Nally, Dennis D Murphy, and John P Fay. 2005. <span>“Using Indicator Species to Predict Species Richness of Multiple Taxonomic Groups.”</span> <em>Conservation Biology</em> 19 (4): 1125–37.
</div>
<div id="ref-fox2017generating" class="csl-entry" role="listitem">
Fox, Helen E, Megan D Barnes, Gabby N Ahmadia, Grace Kao, Louise Glew, Kelly Haisfield, Nur Ismu Hidayat, et al. 2017. <span>“Generating Actionable Data for Evidence-Based Conservation: The Global Center of Marine Biodiversity as a Case Study.”</span> <em>Biological Conservation</em> 210: 299–309.
</div>
<div id="ref-fraixedas2020state" class="csl-entry" role="listitem">
Fraixedas, Sara, Andreas Lindén, Markus Piha, Mar Cabeza, Richard Gregory, and Aleksi Lehikoinen. 2020. <span>“A State-of-the-Art Review on Birds as Indicators of Biodiversity: Advances, Challenges, and Future Directions.”</span> <em>Ecological Indicators</em> 118: 106728.
</div>
<div id="ref-furness2013birds" class="csl-entry" role="listitem">
Furness, Robert W, and Jeremy JD Greenwood. 2013. <em>Birds as Monitors of Environmental Change</em>. Springer Science &amp; Business Media.
</div>
<div id="ref-garland2020acoustic" class="csl-entry" role="listitem">
Garland, Laura, Andrew Crosby, Richard Hedley, Stan Boutin, and Erin Bayne. 2020. <span>“Acoustic Vs. Photographic Monitoring of Gray Wolves (Canis Lupus): A Methodological Comparison of Two Passive Monitoring Techniques.”</span> <em>Canadian Journal of Zoology</em> 98 (3): 219–28.
</div>
<div id="ref-gregory2003using" class="csl-entry" role="listitem">
Gregory, Richard D, David Noble, Rob Field, John Marchant, M Raven, and DW Gibbons. 2003. <span>“Using Birds as Indicators of Biodiversity.”</span> <em>Ornis Hungarica</em> 12 (13): 11–24.
</div>
<div id="ref-hallgren2016biodiversity" class="csl-entry" role="listitem">
Hallgren, Willow, Linda Beaumont, Andrew Bowness, Lynda Chambers, Erin Graham, Hamish Holewa, Shawn Laffan, et al. 2016. <span>“The Biodiversity and Climate Change Virtual Laboratory: Where Ecology Meets Big Data.”</span> <em>Environmental Modelling &amp; Software</em> 76: 182–86.
</div>
<div id="ref-hampton2013big" class="csl-entry" role="listitem">
Hampton, Stephanie E, Carly A Strasser, Joshua J Tewksbury, Wendy K Gram, Amber E Budden, Archer L Batcheller, Clifford S Duke, and John H Porter. 2013. <span>“Big Data and the Future of Ecology.”</span> <em>Frontiers in Ecology and the Environment</em> 11 (3): 156–62.
</div>
<div id="ref-hobson2002acoustic" class="csl-entry" role="listitem">
Hobson, Keith A, Robert S Rempel, Hamilton Greenwood, Brian Turnbull, and Steven L Van Wilgenburg. 2002. <span>“Acoustic Surveys of Birds Using Electronic Recordings: New Potential from an Omnidirectional Microphone System.”</span> <em>Wildlife Society Bulletin</em>, 709–20.
</div>
<div id="ref-huus2025hawkears" class="csl-entry" role="listitem">
Huus, Jan, Kevin G Kelly, Erin M Bayne, and Elly C Knight. 2025. <span>“HawkEars: A Regional, High-Performance Avian Acoustic Classifier.”</span> <em>Ecological Informatics</em> 87: 103122.
</div>
<div id="ref-jetz2012global" class="csl-entry" role="listitem">
Jetz, Walter, Gavin H Thomas, Jeffery B Joy, Klaas Hartmann, and Arne O Mooers. 2012. <span>“The Global Diversity of Birds in Space and Time.”</span> <em>Nature</em> 491 (7424): 444–48.
</div>
<div id="ref-jiguet2007climate" class="csl-entry" role="listitem">
Jiguet, Frédéric, ANNE-SOPHIE GADOT, Romain Julliard, Stuart E Newson, and Denis Couvet. 2007. <span>“Climate Envelope, Life History Traits and the Resilience of Birds Facing Global Change.”</span> <em>Global Change Biology</em> 13 (8): 1672–84.
</div>
<div id="ref-kahl2021birdnet" class="csl-entry" role="listitem">
Kahl, Stefan, Connor M Wood, Maximilian Eibl, and Holger Klinck. 2021. <span>“BirdNET: A Deep Learning Solution for Avian Diversity Monitoring.”</span> <em>Ecological Informatics</em> 61: 101236.
</div>
<div id="ref-kartez2008information" class="csl-entry" role="listitem">
Kartez, Jack D, and Molly P Casto. 2008. <span>“Information into Action: Biodiversity Data Outreach and Municipal Land Conservation.”</span> <em>Journal of the American Planning Association</em> 74 (4): 467–80.
</div>
<div id="ref-kim2021ecobank" class="csl-entry" role="listitem">
Kim, Hyun Woo, Sungsoo Yoon, Mokyoung Kim, Manseok Shin, Heenam Yoon, and Kidong Kim. 2021. <span>“EcoBank: A Flexible Database Platform for Sharing Ecological Data.”</span> <em>Biodiversity Data Journal</em> 9: e61866.
</div>
<div id="ref-kush2020fair" class="csl-entry" role="listitem">
Kush, Rebecca Daniels, D Warzel, Maura A Kush, Alexander Sherman, Eileen A Navarro, R Fitzmartin, Frank Pétavy, et al. 2020. <span>“FAIR Data Sharing: The Roles of Common Data Elements and Harmonization.”</span> <em>Journal of Biomedical Informatics</em> 107: 103421.
</div>
<div id="ref-lepage2014avibase" class="csl-entry" role="listitem">
Lepage, Denis, Gaurav Vaidya, and Robert Guralnick. 2014. <span>“Avibase–a Database System for Managing and Organizing Taxonomic Concepts.”</span> <em>ZooKeys</em>, no. 420: 117.
</div>
<div id="ref-macphail2024audio" class="csl-entry" role="listitem">
MacPhail, Alexander G, Daniel A Yip, Elly C Knight, Richard Hedley, Michelle Knaggs, Julia Shonfield, Emily Upham-Mills, and Erin M Bayne. 2024. <span>“Audio Data Compression Affects Acoustic Indices and Reduces Detections of Birds by Human Listening and Automated Recognisers.”</span> <em>Bioacoustics</em> 33 (1): 74–90.
</div>
<div id="ref-mekonen2017birds" class="csl-entry" role="listitem">
Mekonen, Sefi. 2017. <span>“Birds as Biodiversity and Environmental Indicator.”</span> <em>Indicator</em> 7 (21).
</div>
<div id="ref-michel2020metrics" class="csl-entry" role="listitem">
Michel, Nicole L, Curtis Burkhalter, Chad B Wilsey, Matt Holloran, Alison Holloran, and Gary M Langham. 2020. <span>“Metrics for Conservation Success: Using the <span>‘Bird-Friendliness Index’</span> to Evaluate Grassland and Aridland Bird Community Resilience Across the Northern Great Plains Ecosystem.”</span> <em>Diversity and Distributions</em> 26 (12): 1687–1702.
</div>
<div id="ref-morante2017appraisal" class="csl-entry" role="listitem">
Morante-Filho, José Carlos, and Deborah Faria. 2017. <span>“An Appraisal of Bird-Mediated Ecological Functions in a Changing World.”</span> <em>Tropical Conservation Science</em> 10: 1940082917703339.
</div>
<div id="ref-nathan2022big" class="csl-entry" role="listitem">
Nathan, Ran, Christopher T Monk, Robert Arlinghaus, Timo Adam, Josep Alós, Michael Assaf, Henrik Baktoft, et al. 2022. <span>“Big-Data Approaches Lead to an Increased Understanding of the Ecology of Animal Movement.”</span> <em>Science</em> 375 (6582): eabg1780.
</div>
<div id="ref-newman2007aquatic" class="csl-entry" role="listitem">
Newman, Scott H, Aleksei Chmura, Kathy Converse, A Marm Kilpatrick, Nikkita Patel, Emily Lammers, and Peter Daszak. 2007. <span>“Aquatic Bird Disease and Mortality as an Indicator of Changing Ecosystem Health.”</span> <em>Marine Ecology Progress Series</em> 352: 299–309.
</div>
<div id="ref-niemi1997critical" class="csl-entry" role="listitem">
Niemi, Gerald J, Joann M Hanowski, Ann R Lima, Tom Nicholls, and Norm Weiland. 1997. <span>“A Critical Analysis on the Use of Indicator Species in Management.”</span> <em>The Journal of Wildlife Management</em>, 1240–52.
</div>
<div id="ref-o2008picture" class="csl-entry" role="listitem">
O’Brien, Timothy G, and Margaret F Kinnaird. 2008. <span>“A Picture Is Worth a Thousand Words: The Application of Camera Trapping to the Study of Birds.”</span> <em>Bird Conservation International</em> 18 (S1): S144–62.
</div>
<div id="ref-orme2006global" class="csl-entry" role="listitem">
Orme, C David L, Richard G Davies, Valerie A Olson, Gavin H Thomas, Tzung-Su Ding, Pamela C Rasmussen, Robert S Ridgely, et al. 2006. <span>“Global Patterns of Geographic Range Size in Birds.”</span> <em>PLoS Biology</em> 4 (7): e208.
</div>
<div id="ref-perkel2019ways" class="csl-entry" role="listitem">
Perkel, Jeffrey M. 2019. <span>“Ways to Avoid a Data-Storage Disaster.”</span> <em>Nature</em> 568 (7750): 131–32.
</div>
<div id="ref-peters2014harnessing" class="csl-entry" role="listitem">
Peters, Debra PC, Kris M Havstad, Judy Cushing, Craig Tweedie, Olac Fuentes, and Natalia Villanueva-Rosales. 2014. <span>“Harnessing the Power of Big Data: Infusing the Scientific Method with Machine Learning to Transform Ecology.”</span> <em>Ecosphere</em> 5 (6): 1–15.
</div>
<div id="ref-pollet2025technological" class="csl-entry" role="listitem">
Pollet, Ingrid L, Alexa Arnyek, Julia Ellen Baak, Rikki Clark, Jacob Comeau-Ouellette, Asha C Grewal, Sarah E Gutowsky, et al. 2025. <span>“Technological Advancements: A Global Review of the Use of Camera Technology in Wildlife Research.”</span> <em>Environmental Reviews</em>, no. ja.
</div>
<div id="ref-randler2018distance" class="csl-entry" role="listitem">
Randler, Christoph, and Nadine Kalb. 2018. <span>“Distance and Size Matters: A Comparison of Six Wildlife Camera Traps and Their Usefulness for Wild Birds.”</span> <em>Ecology and Evolution</em> 8 (14): 7151–63.
</div>
<div id="ref-reif2013long" class="csl-entry" role="listitem">
Reif, Jiřı́. 2013. <span>“Long-Term Trends in Bird Populations: A Review of Patterns and Potential Drivers in North America and Europe.”</span> <em>Acta Ornithologica</em> 48 (1): 1–16.
</div>
<div id="ref-sekercioglu2012bird" class="csl-entry" role="listitem">
Sekercioglu, Cagan H. 2012. <span>“Bird Functional Diversity and Ecosystem Services in Tropical Forests, Agroforests and Agricultural Areas.”</span> <em>Journal of Ornithology</em> 153 (Suppl 1): 153–61.
</div>
<div id="ref-shin2015ecological" class="csl-entry" role="listitem">
Shin, Dong-Hee, and Min Jae Choi. 2015. <span>“Ecological Views of Big Data: Perspectives and Issues.”</span> <em>Telematics and Informatics</em> 32 (2): 311–20.
</div>
<div id="ref-shonfield2017autonomous" class="csl-entry" role="listitem">
Shonfield, Julia, and Erin M Bayne. 2017. <span>“Autonomous Recording Units in Avian Ecological Research: Current Use and Future Applications.”</span> <em>Avian Conservation &amp; Ecology</em> 12 (1).
</div>
<div id="ref-sitters2016bird" class="csl-entry" role="listitem">
Sitters, Holly, Julian Di Stefano, Fiona Christie, Matthew Swan, and Alan York. 2016. <span>“Bird Functional Diversity Decreases with Time Since Disturbance: Does Patchy Prescribed Fire Enhance Ecosystem Function?”</span> <em>Ecological Applications</em> 26 (1): 115–27.
</div>
<div id="ref-smits2013avian" class="csl-entry" role="listitem">
Smits, Judit EG, and Kimberly J Fernie. 2013. <span>“Avian Wildlife as Sentinels of Ecosystem Health.”</span> <em>Comparative Immunology, Microbiology and Infectious Diseases</em> 36 (3): 333–42.
</div>
<div id="ref-solymos2013calibrating" class="csl-entry" role="listitem">
Sólymos, Péter, Steven M Matsuoka, Erin M Bayne, Subhash R Lele, Patricia Fontaine, Steve G Cumming, Diana Stralberg, Fiona KA Schmiegelow, and Samantha J Song. 2013. <span>“Calibrating Indices of Avian Density from Non-Standardized Survey Data: Making the Most of a Messy Situation.”</span> <em>Methods in Ecology and Evolution</em> 4 (11): 1047–58.
</div>
<div id="ref-stanton2016flexible" class="csl-entry" role="listitem">
Stanton, Jessica C, Brice X Semmens, Patrick C McKann, Tom Will, and Wayne E Thogmartin. 2016. <span>“Flexible Risk Metrics for Identifying and Monitoring Conservation-Priority Species.”</span> <em>Ecological Indicators</em> 61: 683–92.
</div>
<div id="ref-stephenson2020inventory" class="csl-entry" role="listitem">
Stephenson, PJ, and Carrie Stengel. 2020. <span>“An Inventory of Biodiversity Data Sources for Conservation Monitoring.”</span> <em>PLoS One</em> 15 (12): e0242923.
</div>
<div id="ref-temple1989bird" class="csl-entry" role="listitem">
Temple, Stanley A, and John A Wiens. 1989. <span>“Bird Populations and Environmental Changes: Can Birds Be Bio-Indicators?”</span> <em>American Birds</em> 43 (2): 14.
</div>
<div id="ref-yip2017experimentally" class="csl-entry" role="listitem">
Yip, Daniel A, Lionel Leston, Erin M Bayne, Péter Sólymos, and Alison Grover. 2017. <span>“Experimentally Derived Detection Distances from Audio Recordings and Human Observers Enable Integrated Analysis of Point Count Data.”</span> <em>Avian Conservation and Ecology</em> 12 (1): 11.
</div>
<div id="ref-zakaria2005comparison" class="csl-entry" role="listitem">
Zakaria, Mohamed, Puan Chong Leong, and Muhammad Ezhar Yusuf. 2005. <span>“Comparison of Species Composition in Three Forest Types: Towards Using Bird as Indicator of Forest Ecosystem Health.”</span> <em>Journal of Biological Sciences</em> 5 (6): 734–37.
</div>
<div id="ref-zhang2009efficiently" class="csl-entry" role="listitem">
Zhang, Jianting, Michael Gertz, and Le Gruenwald. 2009. <span>“Efficiently Managing Large-Scale Raster Species Distribution Data in PostgreSQL.”</span> In <em>Proceedings of the 17th ACM SIGSPATIAL International Conference on Advances in Geographic Information Systems</em>, 316–25.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"selector":".lightbox","openEffect":"zoom","loop":false,"descPosition":"bottom","closeEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>